### ISSUE TITLE
Connection reset error

### ISSUE THREAD

**Issue by mzfr** [mushorg/snare#244]
So I installed snare and then tanner. Both runs without any error but when I am trying to view the website it's showing connection reset error in the browser.

* To start snare I ran: `➜ sudo python bin/snare --port 8080 --page-dir example.com --tanner 127.0.0.1`

* To start tanner I ran: `sudo tanner`

When Tried to open `127.0.0.1:8080` I am seeing `Connection Reset`.

I tried to view the `snare.err` but it was empty. In `snare.log` I found the following line:

```
2020-02-26 00:47:07 DEBUG:asyncio:__init__: Using selector: EpollSelector
2020-02-26 00:47:07 DEBUG:git.cmd:execute: Popen(['git', 'fetch', '-v', 'origin'], cwd=/home/snare, universal_newlines=True, shell=None, istream=None)
2020-02-26 00:47:09 DEBUG:git.cmd:wait: AutoInterrupt wait stderr: b'fatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.'
```

So I started snare with the `--auto-update false` option and still, the problem persists and in `snare.log` all I see is 

```
2020-02-26 00:42:25 DEBUG:asyncio:__init__: Using selector: EpollSelector
2020-02-26 00:42:31 DEBUG:asyncio:__init__: Using selector: EpollSelector
2020-02-26 00:42:41 DEBUG:asyncio:__init__: Using selector: EpollSelector
```

**Comment by afeena** [comment]
@mzfr I tried to run snare/tanner on the same server and it works for me. I can't reproduce your problem

Check please also tanner logs

**Comment by mzfr** [comment]
Tanner logs looks good.

There is nothing in the "tanner.err" and tanner.log are fine I mean nothing suspicious in there.

**Comment by mzfr** [comment]
I am looking into this and I noticed something

If I am running snare & tanner locally it never calls `snare.server:handle_request` function even when I try to visit the website in the browser.

But when I am running them with docker it seems to work.

**Comment by afeena** [comment]
@mzfr so you mean you can't reach website in browser?

**Comment by mzfr** [comment]
Yes for some reason it doesn't work locally but works with docker.

**Comment by afeena** [comment]
@mzfr I have exactly the same setup: snare and tanner on the same server and I use --tanner 127.0.0.1 and everything works fine. You can try to use debugger in snare to check what is happening

**Comment by nitish81299** [comment]
I am also facing the same issue. I made a venv, installed the requirements and then cloned example.com

then I used

```
sudo snare --port 80 --page-dir example.com --host-ip 0.0.0.0
```
to run snare, but when I visit the url I get connection reset error.

@afeena @mzfr any suggestions?

**Comment by afeena** [comment]
@nitish81299 probably some issues with default tanner instance. I tried it now and it works, but you can also try with your own tanner instance

**Comment by mzfr** [comment]
Okay so this is because of python version. Everything works in python3.5

**Comment by afeena** [comment]
@mzfr And which one you have used before?

**Comment by mzfr** [comment]
Python3.8

**Comment by afeena** [comment]
@mzfr thanks for the info! I haven't tested anything with python 3.8 yet 😄

**Comment by mzfr** [comment]
oh, my default python version is set to `3.8.1` so my venv was also of that version.

I tested it with python3.5 version and it worked.

Also, the issue is with aiohttp library(I guess), because as I said in [this](https://github.com/mushorg/snare/issues/244#issuecomment-592922447) comment the `async def handle_request` wasn't called so I think maybe aiohttp fails to start a server.

**Comment by Fosocles** [comment]
Python 3.8 here and I get the same error when I visit my cloned website. Using dockers too.

**Comment by mzfr** [comment]
Docker actually worked fine for me. But yeah python 3.8 was causing the issue so I shifted to 3.5 and everything worked

**Comment by Fosocles** [comment]
@mzfr I was using `alpine:latest` which has Python3.8 hence my issue. Fixed it by installing `alpine:3.10`.

**Comment by Fosocles** [comment]
I think this can be closed now?
